core/app/models/workarea/checkout.rb

```diff
@@ -161,7 +161,13 @@ def continue_as(user)
     # @return [self]
     #
     def update(params = {})
-      steps.each { |s| s.new(self).update(params) }
+      steps.all? do |step|
+        result = step.new(self).update(params)
+
+        next true if result.nil?
```

Comment:
this right here allows for steps for which we aren't passing params to be "skipped" when determining whether the update failed or not. otherwise, it would probably just fail every time since `#update` would be returning `nil` for many of these steps.